apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1679,6 +1684,39 @@ adjust_batch_time(Id, RequestTimeout, BatchTime0) ->
     end,
     BatchTime.
 
+%% The request timeout should be greater than the health check
+%% timeout, health timeout defines how often the buffer worker tries
+%% to unblock. If request timeout is <= health check timeout and the
+%% buffer worker is ever blocked, than all queued requests will
+%% basically fail without being attempted.
```

Comment:
It's actually not dependent on being used by a bridge or other things.  

`request_timeout` and `health_check_interval` are used by all buffer workers to deal with request expiration and both health check / resuming from blocked state, respectively. 

So I think it doesn't relate to bridges specifically.  If the user wants to preserve more messages, then they should tune those parameters, possibly increasing request timeout.

Reply:
My point was that a desire _not to preserve messages_ if a resource is down might be reasonable (e.g., for cases other than bridges), but after the change, that will inevitably lead to the alarm.

Reply:
Now I think I understand the issue and I agree: there may be situations where having `request_timeout < health_check_timeout` might be desirable.

I think the root issue is that `resume_interval` is not exposed as a configurable parameter and it's tied to (is the same as) `health_check_interval`.

So I thought of 2 alternative solutions to raising an alarm:

1. Automatically adjust `resume_interval` to be the minimum of either `health_check_interval` and `request_timeout / 2`.
2. Just log a warning suggesting that there might be an inconsistent configuration.
3. Expose `resume_interval` as a config and make the same alarm logic between `request_timeout` and `resume_interval` instead.

I think (1) is less noisy and might resolve the issue.  If the request timeout is too short, however, it might stress the resource more than desirable.

Would you agree, @zmstone ?

Reply:
I had another idea of automatically setting the default `resume_interval` and discussed it with @zmstone .

@savonarola could you please have another look?